Compile error in Node V12.*

We get the following error while installing the package>

`gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/ayhan/devtools/node-12.4.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:200:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Linux 4.15.0-52-generic
gyp ERR! command "/home/ayhan/devtools/node-12.4.0/bin/node" "/home/ayhan/devtools/node-12.4.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/ayhan/projects/matriks/dev/work/nnw/node_modules/nanomsg
gyp ERR! node -v v12.4.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
n`

Use Vue-Cli install nanomsg4.0.2 .My nodejs version is v12.0.0

``..\src\node_nanomsg.cc(102): error C2664: 'v8::String::Utf8Value::Utf8Value(const v8::String::Utf8Value &)': cannot convert argument 1 from 'v8::Local<v8::Value>' to 'const v8::String::Utf8Value &' [F:\ali-project\Kratos\node_modules\nanomsg\build\node_nanomsg.vcxproj]
  ..\src\node_nanomsg.cc(102): note: Reason: cannot convert from 'v8::Local<v8::Value>' to 'const v8::String::Utf8Value'
  ..\src\node_nanomsg.cc(102): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
f:\ali-project\kratos\node_modules\nanomsg\node_modules\nan\nan_object_wrap.h(24): error C2039: 'IsNearDeath': is not a member of 'Nan::Persistent<v8::Object,v8::NonCopyablePersistentTraits<T>>' [F:\ali-project\Kratos\node_modules\nanomsg\build\node_nanomsg.vcxproj]
          with
          [
              T=v8::Object
          ] (compiling source file ..\src\poll_ctx.cc)
  F:\ali-project\Kratos\node_modules\nanomsg\node_modules\nan\nan.h(1579): note: see declaration of 'Nan::Persistent<v8::Object,v8::NonCopyablePersistentTraits<T>>'
          with
          [
              T=v8::Object
          ] (compiling source file ..\src\poll_ctx.cc)
f:\ali-project\kratos\node_modules\nanomsg\node_modules\nan\nan_object_wrap.h(67): warning C4996: 'v8::PersistentBase<v8::Object>::MarkIndependent': was declared deprecated (compiling source file ..\src\poll_ctx.cc) [F:\ali-project\Kratos\node_modules\nanomsg\build\node_nanomsg.vcxproj]
  c:\users\bin\.node-gyp\12.0.0\include\node\v8.h(563): note: see declaration of 'v8::PersistentBase<v8::Object>::MarkIndependent' (compiling source file ..\src\poll_ctx.cc)
f:\ali-project\kratos\node_modules\nanomsg\node_modules\nan\nan_object_wrap.h(124): error C2039: 'IsNearDeath': is not a member of 'Nan::Persistent<v8::Object,v8::NonCopyablePersistentTraits<T>>' [F:\ali-project\Kratos\node_modules\nanomsg\build\node_nanomsg.vcxproj]
          with
          [
              T=v8::Object
          ] (compiling source file ..\src\poll_ctx.cc)
  F:\ali-project\Kratos\node_modules\nanomsg\node_modules\nan\nan.h(1579): note: see declaration of 'Nan::Persistent<v8::Object,v8::NonCopyablePersistentTraits<T>>'
          with
          [
              T=v8::Object
          ] (compiling source file ..\src\poll_ctx.cc)
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\bin\AppData\Roaming\nvm\v12.0.0\node_modules\npm\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:196:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:256:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\bin\\AppData\\Roaming\\nvm\\v12.0.0\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd F:\ali-project\Kratos\node_modules\nanomsg
gyp ERR! node -v v12.0.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok``

--- a/package.json
+++ b/package.json
@@ -5,7 +5,7 @@
   "main": "lib/index.js",
   "dependencies": {
     "bindings": "1.2.1",
-    "nan": "2.8"
+    "nan": "2.14.0"
   },
   "devDependencies": {
     "buffer-equals-polyfill": "1.0.0",

--- a/src/node_nanomsg.cc
+++ b/src/node_nanomsg.cc
@@ -230,7 +230,7 @@ public:
 
     Local<Value> argv[] = { Nan::New<Number>(err) };
 
-    callback->Call(1, argv);
+    callback->Call(1, argv, async_resource);
   };
 
 private:

--- a/src/node_nanomsg.cc
+++ b/src/node_nanomsg.cc
@@ -64,7 +64,7 @@ NAN_METHOD(Chan) {
   int s = Nan::To<int>(info[0]).FromJust();
   int level = NN_SUB;
   int option = Nan::To<int>(info[1]).FromJust();
-  String::Utf8Value str(info[2]);
+  Nan::Utf8String str(info[2]);
 
   info.GetReturnValue().Set(
       Nan::New<Number>(nn_setsockopt(s, level, option, *str, str.length())));
@@ -72,14 +72,14 @@ NAN_METHOD(Chan) {
 
 NAN_METHOD(Bind) {
   int s = Nan::To<int>(info[0]).FromJust();
-  String::Utf8Value addr(info[1]);
+  Nan::Utf8String addr(info[1]);
 
   info.GetReturnValue().Set(Nan::New<Number>(nn_bind(s, *addr)));
 }
 
 NAN_METHOD(Connect) {
   int s = Nan::To<int>(info[0]).FromJust();
-  String::Utf8Value addr(info[1]);
+  Nan::Utf8String addr(info[1]);
 
   info.GetReturnValue().Set(Nan::New<Number>(nn_connect(s, *addr)));
 }
@@ -99,7 +99,7 @@ NAN_METHOD(Send) {
     info.GetReturnValue().Set(Nan::New<Number>(nn_send(
         s, node::Buffer::Data(info[1]), node::Buffer::Length(info[1]), flags)));
   } else {
-    String::Utf8Value str(info[1]);
+    Nan::Utf8String str(info[1]);
     info.GetReturnValue().Set(
         Nan::New<Number>(nn_send(s, *str, str.length(), flags)));
   }

